Px.Utils/Validation/ContentValidation/ContentValidator.cs

```diff
@@ -0,0 +1,141 @@
+﻿using PxUtils.PxFile;
+using PxUtils.Validation.SyntaxValidation;
+using System.Collections.Generic;
+using System.Text;
+
+namespace PxUtils.Validation.ContentValidation
+{
+    /// <summary>
+    /// Provides methods for validating the content of Px file metadata
+    /// <param name="filename">Name of the Px file</param>
+    /// <param name="encoding"> Encoding of the Px file</param>
+    /// </summary>
+    public class ContentValidator(string filename, Encoding encoding)
+    {
+        public string Filename { get; } = filename;
```

Comment:
Check which of these need to actually be public. Change all to private that can be and hide the setters from the rest. Also the reference types should be readonly.